.github/workflows/build-release-binary.yml

```diff
@@ -129,3 +154,29 @@ jobs:
           asset_path: ./${{ steps.create-archive-name.outputs.archive }}
           asset_name: ${{ steps.create-archive-name.outputs.archive }}
           asset_content_type: application/gzip
+
+      - name: Set up QEMU
+        if: matrix.os != 'windows-latest' && matrix.os != 'macos-latest'
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        if: matrix.os != 'windows-latest' && matrix.os != 'macos-latest'
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login into github registry
+        uses: docker/login-action@v1.10.0
+        if: matrix.os != 'windows-latest' && matrix.os != 'macos-latest'
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and publish docker image
+        if: matrix.os != 'windows-latest' && matrix.os != 'macos-latest'
+        run: |
```

Comment:
Instead of making all of these conditional, would it work to:

1. Have a separate job after this
2. With a dependency on the first one
3. Download the artifacts from the above job

The benefits I see is that:

- Build matrix gets simpler because all the new variables are only concerned with docker, which would be in a new job.
- The job declaration gets easier because we don't need to exclude it for macos and windows.

Reply:
I would assume so but I haven't tried this. 
Sounds like an "easy first ticket" @klochowicz ?

Reply:
I'll postpone this to the future #285

Reply:
> I'll postpone this to the future #285

Thanks for creating a ticket. It is a bit hard to understand out of context so some re-wording would be nice :)